presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -195,16 +202,23 @@ private boolean joinCurrentPosition()
     {
         // while we have a position on lookup side to join against...
         while (joinPosition >= 0) {
-            pageBuilder.declarePosition();
-
-            // write probe columns
-            probe.appendTo(pageBuilder);
-
-            // write build columns
-            lookupSource.appendTo(joinPosition, pageBuilder, probe.getOutputChannelCount());
+            lookupPositionsConsidered++;
+            if (lookupSource.isJoinPositionEligible(joinPosition, probe.getPosition(), probe.getPage())) {
```

Comment:
To be honest it would be much easier to write some sql with https://github.com/prestodb/presto/pull/5809 . Adding some filter function to `BenchmarkHashBuildAndJoinOperators` benchmarks would probably be to mush of a hassle. 

However here I'm only slightly concerned with calls to `isJoinPositionEligible` and not with actual filter function, so you could maybe only add a test case to `BenchmarkHashBuildAndJoinOperators` for build table that has duplicated (10?) keys (if it's not already there)?